Add on_response callback and remove waiting_time

dgb opened the pull request:
Adds an `on_response` callback to `Fb::HTTPRequest` and removes
`waiting_time`. This can be used (as shown in the README) to implement
`waiting_time`'s behavior in the consuming application, while allowing
further flexibility.

dgb committed:
Add on_response callback and remove waiting_time

Adds an `on_response` callback to `Fb::HTTPRequest` and removes
`waiting_time`. This can be used (as shown in the README) to implement
`waiting_time`'s behavior in the consuming application, while allowing
further flexibility.

dgb committed:
Use class variable for on_response

Since the HTTPRequest is subclassed, we want this to be configurable
across all subclasses of HTTPRequest as well.

kangkyu commented on README.md:
```diff
@@ -20,14 +20,24 @@ Fb::Support provides:
+  Librato.measure 'fb.call_count', parsed['call_count']
+  Librato.measure 'fb.total_cputime', parsed['total_cputime']
+  Librato.measure 'fb.total_time', parsed['total_time']
+  if parsed['call_count'] > 85
```
we still can use `rate_limiting_header` method here?

dgb replied:
I think we'd need to pass the request object back in order to do that. Do you think the callback should be something like `lambda {|request, response| ... }`?

kangkyu replied:
Then this part of README can be like this?
```rb
Fb::HTTPRequest.on_response = lambda do |request, response|
  usage = request.rate_limiting_header
  Librato.measure 'fb.call_count', usage['call_count']
  Librato.measure 'fb.total_cputime', usage['total_cputime']
  Librato.measure 'fb.total_time', usage['total_time']
  if usage.values.any? {|value| value > 85 }
    sleep 180
  end
end

dgb replied:
Oh right, let me update the README

dgb committed:
Pass the request along too

Passes the request along to the callback as well, since it has things
like `rate_limiting_header`.

dgb committed:
Update README
